### PR TITLE
Streaming files

### PR DESCRIPTION
**Pull request by mfleader**
Add stream endpoint to server, so that it can write file chunks directly to a file. This allows the server to receive files that are larger than its host's memory.

### CODE REVIEW

File: test/cm/configure_fedora34.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#! /bin/bash
+
+# sudo dnf install -y podman git perl-open.noarch python3-devel gcc libpq-devel python3.9-devel
+# git clone https://github.com/mfleader/snappy-data-server.git
+# sudo chown -R vagrant snappy-data-server
+# sudo chgrp -R vagrant snappy-data-server
```
Comment (Harshith-umesh) [inline test/cm/configure_fedora34.sh]: delete these lines ?

File: requirements.txt [hunk 2]
```diff
@@ -1,36 +1,49 @@
...
-uvicorn==0.13.1
-werkzeug==1.0.1; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'
+six==1.16.0; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
+sqlalchemy==1.3.24; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3'
+starlette==0.14.2; python_version >= '3.6'
+tenacity==7.0.0
+typing-extensions==3.10.0.0
+uvicorn==0.13.4
+uvloop==0.15.2
+watchgod==0.7
```
Comment (Harshith-umesh) [inline requirements.txt]: are all these requirements needed ?

File: test/cm/configure_fedora34.sh (hunk 1 above)
Comment (mfleader) [reply test/cm/configure_fedora34.sh]: I can remove `configure_fedora34.sh`. I added it to the repo as a reflex.

File: requirements.txt (hunk 2 above)
Comment (mfleader) [reply requirements.txt]: Yes, except `watchgod` should be a development dependency. 

`uvloop` and `httptools`, as suggested by [uvicorn's docs](https://www.uvicorn.org/#quickstart) should bring us performance improvements in serving concurrent requests.

**Comment by Harshith-umesh** [comment]
LGTM!

**Pull request merged**